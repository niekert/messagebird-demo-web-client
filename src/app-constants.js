// Status codes
export const INITIAL = 'INITIAL';
export const PENDING = 'PENDING';
export const OK = 'OK';
export const ERROR = 'ERROR';
