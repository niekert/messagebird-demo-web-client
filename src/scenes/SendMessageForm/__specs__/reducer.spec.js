import { INITIAL } from 'app-constants';
import reduce from '../reducer';
import {
  SEND_MESSAGE,
  SEND_MESSAGE_ERROR,
  SEND_MESSAGE_SUCCESS,
} from '../actions';

const initialState = {
  status: INITIAL,
  error: null,
};

describe('sendMessageReducer', () => {
  test(`${SEND_MESSAGE}`, () => {
    const result = reduce(initialState, {
      type: SEND_MESSAGE,
    });

    expect(result).toMatchSnapshot();
  });

  test(`${SEND_MESSAGE_SUCCESS}`, () => {
    const result = reduce(initialState, {
      type: SEND_MESSAGE_SUCCESS,
      payload: { query: 'mock_query ' },
    });

    expect(result).toMatchSnapshot();
  });

  test(`${SEND_MESSAGE_ERROR}`, () => {
    const result = reduce(initialState, {
      type: SEND_MESSAGE_ERROR,
      payload: {
        error: new Error('something went wrong'),
      },
    });

    expect(result).toMatchSnapshot();
  });
});
