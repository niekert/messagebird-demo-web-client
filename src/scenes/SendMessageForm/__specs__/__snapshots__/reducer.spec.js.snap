// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendMessageReducer SEND_MESSAGE 1`] = `
Object {
  "error": null,
  "status": "PENDING",
}
`;

exports[`sendMessageReducer SEND_MESSAGE_ERROR 1`] = `
Object {
  "error": [Error: something went wrong],
  "status": "ERROR",
}
`;

exports[`sendMessageReducer SEND_MESSAGE_SUCCESS 1`] = `
Object {
  "error": null,
  "status": "OK",
}
`;
